cpp/src/copying/gather.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/table/table_view.hpp>
+
+#include <cudf/table/table.hpp>
+
+#include <memory>
+
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+void gather(table_view source_table, column_view gather_map,
```

Comment:
Detail APIs should have the `gather_map` as an iterator still, right?

Reply:
Currently we have both APIs in `detail`. The version that works with an iterator is defined in `gather.cuh`. Is this OK?